### DISCUSSION TITLE
IClassFixture<T> and test failures

### DISCUSSION THREAD

**Discussion by blacktau** [xunit/xunit#2783] [Question] [answered]
Hi

I'm using an `IClassFixture<MyFixture>` style of test to share an instance of `MyFixture` between my tests in a single class. 
The issue i'm experiencing is that if one of the tests in the class fails the `MyFixture` instance is disposed as part of processing the failure (Assert fails) and all tests scheduled to be run after the failing test also all fail because the `MyFixture` has been disposed. 

Is this expected behaviour? Is there a way to avoid this?

**Comment by bradwilson** [comment, marked as answer]
This is not expected behavior. Can you provide a repro project? If so, please open an issue.

Thanks!

**Reply by blacktau** [reply]
I'll set up a simplified one this weekend - thanks :)
